backend/src/main/java/com/redhat/cloud/notifications/routers/internal/kessel/KesselAssetsMigrationService.java

```diff
@@ -0,0 +1,144 @@
+package com.redhat.cloud.notifications.routers.internal.kessel;
+
+import com.redhat.cloud.notifications.auth.ConsoleIdentityProvider;
+import com.redhat.cloud.notifications.auth.kessel.ResourceType;
+import com.redhat.cloud.notifications.auth.rbac.workspace.WorkspaceUtils;
+import com.redhat.cloud.notifications.config.BackendConfig;
+import com.redhat.cloud.notifications.db.repositories.EndpointRepository;
+import com.redhat.cloud.notifications.db.repositories.StatusRepository;
+import com.redhat.cloud.notifications.models.Endpoint;
+import com.redhat.cloud.notifications.models.Status;
+import io.grpc.stub.StreamObserver;
+import io.quarkus.logging.Log;
+import jakarta.annotation.security.RolesAllowed;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.BadRequestException;
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.Path;
+import org.project_kessel.api.relations.v1beta1.CreateTuplesRequest;
+import org.project_kessel.api.relations.v1beta1.CreateTuplesResponse;
+import org.project_kessel.api.relations.v1beta1.ObjectReference;
+import org.project_kessel.api.relations.v1beta1.ObjectType;
+import org.project_kessel.api.relations.v1beta1.Relationship;
+import org.project_kessel.api.relations.v1beta1.SubjectReference;
+import org.project_kessel.relations.client.RelationTuplesClient;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.redhat.cloud.notifications.Constants.API_INTERNAL;
+
+@ApplicationScoped
+@RolesAllowed(ConsoleIdentityProvider.RBAC_INTERNAL_ADMIN)
+@Path(API_INTERNAL)
+public class KesselAssetsMigrationService {
+    /**
+     * Defines the relation between the subject and the resource when passing
+     * the tuple to Kessel. <a href="https://github.com/RedHatInsights/rbac-config/blob/a806fb03c95959391eceb0b42c7eefd8ae2350ae/configs/prod/schemas/schema.zed#L3">
+     * Reference</a>
+     */
+    public static final String RELATION = "t_workspace";
```

Comment:
The relation should be "workspace" – the t_ prefix is an internal implementation detail in how we are mapping to SpiceDB's schema constructs. Go by the relation that's in KSL – be careful going by what you see in the underlying SpiceDB schema.

Reply:
i.e. this is the relevant line to link to: https://github.com/RedHatInsights/rbac-config/blob/bb2d840a33705e84e0819d5fe697aaa7c87c378d/configs/prod/schemas/src/notifications.ksl#L13

Reply:
Got it, thank you!